pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -325,56 +325,32 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	// sets the v1.NodeNetworkUnavailable condition to False
 	ca.setNetworkCondition(node)
 
-	// nodes in clusters WITHOUT multi-networking are expected to have
-	// only 1 network-interface and 1 alias IPv4 range or/and 1 IPv6 address
-	// multi-network cluster may have 1 interface with multiple alias
-	if len(instance.NetworkInterfaces) == 1 &&
-		(len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 ||
-			ca.cloud.GetIPV6Address(instance.NetworkInterfaces[0]) != nil) {
-		// with 1 alias IPv4 range on single IPv4 or dual stack clusters
-		if len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 {
-			cidrStrings = append(cidrStrings, instance.NetworkInterfaces[0].AliasIpRanges[0].IpCidrRange)
-		}
-		// with 1 IPv6 range on single IPv6 or dual stack cluster
+	// nodes in clusters WITHOUT multi-networking are expected to have only 1 network-interface with 1 alias IP range.
```

Comment:
is this keeping the existing behavior? we can not change the behavior of the cloud_cidr_controller without multinetwork, is a breaking change and there are multiple projects that depend on that

Reply:
Great thanks @aojea! I revert in the second commit, it looks more clean now.